### PR TITLE
Arc support

### PR DESCRIPTION
**Pull request by Capo01**
I have been testing the arc supported branch with a pen potter on an Arduino mega and it appears to work just fine.

### CODE REVIEW

**Comment by Traumflug** [comment]
Thanks a lot for trying!

As Github found out already:
> This branch has conflicts that must be resolved

Next to resolving conflicts, this branch needs more work:
- Wrap the new code in a #define like `SUPPORT_ARC`. Typically printers need no arc support, G-code generators resolve this in a more general way already.
- Add this #define to the configuration files.
- Bonus points for adding it to Configtool, too.
- Wrap arc debug code in `#ifdef DEBUG`.
- Remove this `RESUME_PIN` stuff.
- Remove this E endstop stuff (extruders have no endstop).
- Remove all the commented code.
- Fix intentation (replace tabs by spaces, indentation is 2 spaces).

In short: this code needs some love.
